src/rustllvm/RustWrapper.cpp

```diff
@@ -1442,4 +1442,51 @@ extern "C" LLVMValueRef
 LLVMRustBuildVectorReduceFMax(LLVMBuilderRef B, LLVMValueRef Src, bool NoNaN) {
   return wrap(unwrap(B)->CreateFPMaxReduce(unwrap(Src), NoNaN));
 }
+
+#else
+
+extern "C" LLVMValueRef
+LLVMRustBuildVectorReduceFAdd(LLVMBuilderRef, LLVMValueRef, LLVMValueRef Src) {
+  return Src;
+}
+extern "C" LLVMValueRef
+LLVMRustBuildVectorReduceFMul(LLVMBuilderRef, LLVMValueRef, LLVMValueRef Src) {
+  return Src;
```

Comment:
This is going to lead to broken IR (you'll wind up with vectors where scalars are expected). To truly support earlier LLVM versions, you'll need a polyfill that does the reduction "by hand".